docs/sources/flow/reference/components/prometheus.remote_write.md

```diff
@@ -361,3 +361,72 @@ prometheus.remote_write "default" {
 
 Any labels that start with `__` will be removed before sending to the endpoint.
+
+## Data retention
+
+The `prometheus.remote_write` component reuses the Prometheus Write-Ahead Log
+(WAL) implementation for resiliency against network outages. The component
+buffers the received metrics to a WAL per configured endpoint so that the
+[queue shards][] can flush them to it.
+
+The WAL records metrics in 128MB-files called segments. To avoid having a WAL
+that grows on-disk indefinitely, the component _truncates_ its segments on a
+set interval.
+
+On each truncation, the WAL deletes references to series that are no longer
+present to reduce memory usage, and also _checkpoints_ roughly the oldest
+2/3rds of the segments written to it. A checkpoint means that the WAL only
+keeps track of the unique identifier for each existing metrics series, but
+deletes the actual samples associated with it. If that data has not been pushed
+to remote_write yet, it is lost.
+
+This behavior dictates the data retention for the `prometheus.remote_write`
+component. It also means that it is not possible to directly correlate data
+retention directly to the data age itself, as the truncation logic works on
+_segments_, not the samples themselves. This makes data retention less
+predictable when the component receives a non-consistent rate of data.
+
+The [wal block][] contains some configurable parameters that can be
+used to control the tradeoff between memory usage, disk usage and data
+retention.
+
+The `truncate_frequency` parameter configures the interval at which truncations
+happen. A lower value leads to reduced memory usage, but also provides less
+resiliency to long outages.
+
+When a WAL clean-up starts, the lowest successfully sent timestamp is used to
+determine how much data is safe to remove from the WAL.
+The `min_keepalive_time` controls the minimum age of samples considered for
+removal; no samples more recent than `min_keepalive_time` are removed.
+The `max_keepalive_time` controls the maximum age of samples that can be kept
+in the WAL; samples older than `max_keepalive_time` are forcibly removed.
+
+### In cases of `remote_write` outages
+When the remote write endpoint is unreachable over a period of time, the lowest
+successfully sent timestamp is not updated. In that case, only the
+`min_keepalive_time` and `max_keepalive_time` arguments control the age range
+of data kept in the WAL.
+
+If the remote write outage is longer than the `max_keepalive_time` parameter,
+then the WAL is truncated and the oldest data is lost.
+
+### In cases of intermittent `remote_write` outages
+In case the remote write endpoint is intermittently reachable, the lowest
+succesfully sent timestamp is updated whenever the connection is successful.
+This updates the series' comparison with `min_keepalive_time` and triggers a
+truncation on the `truncate_frequency` interval which will checkpoint roughly
```

Comment:
```suggestion
truncation on the `truncate_frequency` interval which will checkpoint
```

Reply:
This is not an exact number, as it correlates with the number of _segments_, or 128MB-sized files and not the data itself, that's why I put 'roughly' there.

Is there a better way to express that?

Reply:
Yeah, I get what you're saying. Your suggestion reads better now.